src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs

```diff
@@ -256,9 +350,57 @@ private static async Task<Metrics> Get(HttpMessageInvoker client)
         return metrics;
     }
 
+    private static void CreateRequestContentData()
+    {
+        s_useByteArrayContent = !s_options.ContentUnknownLength && (!s_options.ContentFlushAfterWrite || s_options.ContentWriteSize >= s_options.ContentSize); // no streaming
+        if (s_useByteArrayContent)
+        {
+            s_fullChunkCount = 1;
+            s_requestContentData = new byte[s_options.ContentSize];
+            Array.Fill(s_requestContentData, (byte)'a');
+        }
+        else
+        {
+            s_fullChunkCount = s_options.ContentSize / s_options.ContentWriteSize;
+            if (s_fullChunkCount != 0)
+            {
+                s_requestContentData = new byte[s_options.ContentWriteSize];
+                Array.Fill(s_requestContentData, (byte)'a');
+            }
+            int lastChunkSize = s_options.ContentSize % s_options.ContentWriteSize;
+            if (lastChunkSize != 0)
+            {
+                s_requestContentLastChunk = new byte[lastChunkSize];
+                Array.Fill(s_requestContentLastChunk, (byte)'a');
+            }
+        }
+    }
+
+    private static void PrepareCustomHeaders()
+    {
+        foreach (var header in s_options.Header!)
+        {
+            var headerNameValue = header.Split(":", 2, StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+            s_customHeaders.Add((headerNameValue[0], headerNameValue.Length > 1 ? headerNameValue[1] : null));
+        }
+    }
+
     private static HttpRequestMessage CreateRequest(HttpMethod method, Uri uri) =>
         new HttpRequestMessage(method, uri) { Version = s_options.HttpVersion!, VersionPolicy = HttpVersionPolicy.RequestVersionExact };
 
+    private static Task<HttpResponseMessage> SendAsync(HttpMessageInvoker client, HttpRequestMessage request) 
+    {
+        foreach (var header in s_customHeaders)
+        {
+            request.Headers.Remove(header.Name);
```

Comment:
How come this remove is needed?

Reply:
To be able to manually override something from e.g. random generated headers. Do you think we would not need that? Well, we can always add it later if the time shows we *do* need to override things...

Reply:
removed it